libkbfs/folder_branch_ops.go

```diff
@@ -3576,21 +3577,35 @@ func (fbo *folderBranchOps) SetMtime(
 		})
 }
 
-func (fbo *folderBranchOps) syncLocked(ctx context.Context,
-	lState *lockState, file path) (stillDirty bool, err error) {
+// startSyncLocked readies the blocks and other state needed to sync a
+// single file.  It returns:
+//
+// * `doSync`: Whether or not the sync should actually happen.
+// * `stillDirty`: Whether the file should still be considered dirty when
+//   this function returns.  (That is, if `doSync` is false, and `stillDirty`
+//   is true, then the file has outstanding changes but the sync was vetoed for
+//   some other reason.)
+// * `fblock`: the top file block for the file being sync'd.
+// * `lbc`: A local block cache consisting of a dirtied version of the parent
+//   directory for this file.
```

Comment:
Just the parent -- the higher-up parents don't need to change their Children at all, so they're not dirty.